src/main.cc

```diff
@@ -3,21 +3,20 @@
 #include <stdio.h>
 #include <getopt.h>
 #include <string>
+#include "run.hh"
 
 #define BUFSZ 512
 
-/*
- * Logs statistics about the execution of the algorithm and dumps it to a file.
- * To turn off, pass verbosity <= 1
- */
-NullLogger* logger;
+#ifndef _WIN32
+#define _snprintf snprintf
+#endif
 
 int main(int argc, char *argv[]) {
     const char usage[] = "USAGE: %s [-b] "
-        "[-n max_num_nodes] [-r regularization] [-v verbosity] "
-        "-c (1|2|3|4) -p (0|1|2) [-f logging_frequency]"
+        "[-n max_num_nodes] [-r regularization] [-v (rule|label|minor|samples|progress|loud|silent)] "
```

Comment:
If verbosity is using these enums, then why are we checking ints in the code above?

Reply:
Sorry, I'm not sure I understand - what code above are you referring to? There is also a verbosity integer passed to the Logger, which specifies whether a log file should be generated, and how much progress should be printed. The value of the integer depends on the enums (see line 79 of run.cpp).